lib/sms_record.js

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+var actions = require('./actions')
+
+// Keep track of events tied to specific sms number
+module.exports = function (limits, now) {
+
+  now = now || Date.now
+
+  function SMSRecord() {
+    this.xs = []
+  }
+
+  SMSRecord.parse = function (object) {
+    var rec = new SMSRecord()
+    object = object || {}
+    rec.rl = object.rl       // timestamp when the account was rate-limited
+    rec.xs = object.xs || [] // timestamps when sms were sent
```

Comment:
Yes, IIRC exactly this, those properties go straight into memcache as JSON so this was designed to keep the memory pressure down.